File: app/src/main/java/com/android/oleksandrpriadko/demo/loggalitic/FabricConverter.java
package com.android.oleksandrpriadko.demo.loggalitic;

import android.os.Bundle;
import android.support.annotation.NonNull;

import com.android.oleksandrpriadko.loggalitic.analytics.AnalyticsEvent;
import com.android.oleksandrpriadko.loggalitic.analytics.converter.Converter;
import com.crashlytics.android.answers.CustomEvent;

public class FabricConverter extends Converter<CustomEvent> {

    @Override
    public CustomEvent convert(@NonNull AnalyticsEvent analyticsEvent) {
        CustomEvent event = new CustomEvent(analyticsEvent.getName());
        Bundle attrs = analyticsEvent.getAttributes();
        for (String key : attrs.keySet()) {
            if (attrs.get(key) instanceof Number) {
                event.putCustomAttribute(key, (Number) attrs.get(key));
            } else {
                event.putCustomAttribute(key, (String) attrs.get(key));
            }
        }
        return event;
    }

    @Override
    public Class getType() {
        return FabricConverter.class;
    }
}

File: app/src/main/java/com/android/oleksandrpriadko/demo/loggalitic/FabricConverter.kt
package com.android.oleksandrpriadko.demo.loggalitic

import com.android.oleksandrpriadko.core.policy.AnalyticsEvent
import com.android.oleksandrpriadko.loggalitic.analytics.converter.Converter
import com.crashlytics.android.answers.CustomEvent

class FabricConverter : Converter<CustomEvent>() {

    override fun getType(): Class<*> = FabricConverter::class.java

    override fun convert(analyticsEvent: AnalyticsEvent): CustomEvent {
        val event = CustomEvent(analyticsEvent.name)
        val attrs = analyticsEvent.attributes
        for (key in attrs.keySet()) {
            if (attrs.get(key) is Number) {
                event.putCustomAttribute(key, attrs.get(key) as Number)
            } else {
                event.putCustomAttribute(key, attrs.get(key) as String)
            }
        }
        return event
    }
}